scripts/processDiagnosticMessages.ts

```diff
@@ -9,57 +10,55 @@ interface DiagnosticDetails {
 type InputDiagnosticMessageTable = ts.Map<DiagnosticDetails>;
 
 function main(): void {
-    var sys = ts.sys;
+    const sys = ts.sys;
     if (sys.args.length < 1) {
-        sys.write("Usage:" + sys.newLine)
+        sys.write("Usage:" + sys.newLine);
         sys.write("\tnode processDiagnosticMessages.js <diagnostic-json-input-file>" + sys.newLine);
         return;
     }
 
     function writeFile(fileName: string, contents: string) {
-        // TODO: Fix path joining
-        var inputDirectory = inputFilePath.substr(0,inputFilePath.lastIndexOf("/"));
-        var fileOutputPath = inputDirectory + "/" + fileName;
+        const inputDirectory = ts.getDirectoryPath(inputFilePath);
+        const fileOutputPath = ts.combinePaths(inputDirectory, fileName);
         sys.writeFile(fileOutputPath, contents);
     }
 
-    var inputFilePath = sys.args[0].replace(/\\/g, "/");
-    var inputStr = sys.readFile(inputFilePath);
+    const inputFilePath = sys.args[0].replace(/\\/g, "/");
+    const inputStr = sys.readFile(inputFilePath);
 
-    var diagnosticMessagesJson: { [key: string]: DiagnosticDetails } = JSON.parse(inputStr);
-    // Check that there are no duplicates.
-    const seenNames = ts.createMap<true>();

```

Comment:
why was this removed?

Reply:
Object.keys never returns duplicates, the whole block was useless.

Reply:
:D nice catch.